### ISSUE TITLE
[Bug] Cannot update if `PATH` is longer than 8192 characters

### ISSUE THREAD

**Issue by Chaoses-Ib** [ScoopInstaller/Scoop#5874]
#### Current Behavior
When `PATH` is longer than 8192 characters, Scoop cannot run git to update.

#### Expected Behavior
The maximum length of an environment variable is [32767 characters](https://learn.microsoft.com/en-gb/windows/win32/procthread/environment-variables) on Windows. Scoop should work if `PATH` is longer than 8192 characters but less than 32767 characters, or at least print a warning.

#### Additional context/output
```pwsh
> $env:PATH = "${env:USERPROFILE}\scoop\shims;" + 'C:\Windows\System32;'*410
> $env:PATH.length
8229

> git
usage: git [-v | --version] [-h | --help] [-C <path>] [-c <name>=<value>]
           [--exec-path[=<path>]] [--html-path] [--man-path] [--info-path]
...

> scoop install xx
Updating Scoop...
'git' is not recognized as an internal or external command,
operable program or batch file.
Update failed.
Couldn't find manifest for 'xx'.

> scoop update xx 
Updating Scoop...
'git' is not recognized as an internal or external command,
operable program or batch file.
Update failed.
```

#### Possible Solution
8192 characters is the maximum environment variable length accepted by CMD. And Scoop uses CMD to run git here:

https://github.com/ScoopInstaller/Scoop/blob/f93028001fbe5c78cc41f59e3814d2ac8e595724/lib/core.ps1#L131-L138

To relax the length limit to 32767 characters, this usage of CMD needs to be removed. Alternatively, truncating `PATH` and printing a warning if `PATH` is too long would be simpler. If the latter is preferred, I can make a PR.

### System details

**Windows version:** 11

**OS architecture:** 64bit

**PowerShell version:** 7.4.1

#### Scoop Configuration
```json
{
  "last_update": "2024-04-08T18:57:30.2731551+08:00",
  "scoop_branch": "master",
  "scoop_repo": "https://github.com/ScoopInstaller/Scoop"
}
```

**Comment by niheaven** [comment]
It may be partially solved by #5840, and a PR is welcomed always.

**Comment by rashil2000** [comment]
Filing this under #1813
